front/app/components/ProjectAndFolderCards/ProjectAndFolderCards.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useMemo } from 'react';
+
+// components
+import Topbar from './components/Topbar';
+import EmptyContainer from './components/EmptyContainer';
+import ProjectsList from './components/ProjectsList';
+import LoadingBox from './components/LoadingBox';
+import Footer from './components/Footer';
+
+// hooks
+import useAdminPublications, {
+  InputProps as UseAdminPublicationInputProps,
+} from 'hooks/useAdminPublications';
+
+// tracking
+import { trackEventByName } from 'utils/analytics';
+import tracks from './tracks';
+
+// style
+import styled from 'styled-components';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { getAvailableTabs } from './utils';
+
+// typings
+import { IStatusCounts } from 'hooks/useAdminPublicationsStatusCounts';
+import { PublicationTab } from './';
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+export type TLayout = 'dynamic' | 'threecolumns' | 'twocolumns';
+
+export interface BaseProps extends UseAdminPublicationInputProps {
+  showTitle: boolean;
+  layout: TLayout;
+}
+
+interface Props extends BaseProps {
+  currentTab: PublicationTab;
+  statusCounts: IStatusCounts;
+  onChangeAreas: (areas: string[]) => void;
+  onChangeTab: (tab: PublicationTab) => void;
+}
+
+const ProjectAndFolderCards = ({
+  currentTab,
+  statusCounts,
+  showTitle,
+  layout,
+  publicationStatusFilter,
```

Comment:
Don't agree: it is one filter that matches multiple types. 'Filters' seems like it would perform a filtering operation twice (e.g. `data.filter(x => x === condition1).filter(x =? x === condition2)`), but what is happening here is an 'OR' operation (e.g. `array.filter(x = x === condition1 || x === condition2)`. So I am not in favor if renaming it to filters. 

Moreover, I agree `publicationStatusFilter` is not the best naming (I didn't come up with it), but it is in many other places of the codebase too. I think `allowedPublicationStatuses` is better, but then we have to `publicationStatusFilter` to `allowedPublicationStatuses` everywhere in the codebase IMO.

Reply:
If it's an array of values, even if it can just be one value in an array, it should be plural to indicate this. If it is singular, it means it should be a single value. Then publicationStatusFilter type should be PublicationStatus and not PublicationStatus[].

With allowedPublicationStatuses I meant 'allowed as set in the component's prop'. It doesn't need to be changed everywhere. It's just this prop and is separate from the statuses in e.g. useAdminPublications. It's for that reason we're using publicationStatusFilter; because using publicationStatuses everywhere isn't possible due to naming conflicts (+ it's confusing). I'm all for using different names to clearly indicate where it's coming from or used. Check e.g. how I used something like 'statusFiltersPerTab' in the review PR. Makes it much easier to remember where the value is coming from. :)

Reply:
> Don't agree: it is one filter that matches multiple publication statuses. 'Filters' seems like it would perform a filtering operation twice (e.g. data.filter(x => x === condition1).filter(x => x === condition2)), but what is happening here is an 'OR' operation (e.g. array.filter(x = x === condition1 || x === condition2). So I am not in favor if renaming it to filters.

If it's an array of potentially multiple values the name should be plural to indicate this, even if it can be an array of one item. It's how we do it everywhere in the codebase, right?

> I agree publicationStatusFilter is not the best naming (I didn't come up with it), but it is used in many other places in the codebase. I think allowedPublicationStatuses is better, but then we have to publicationStatusFilter to allowedPublicationStatuses everywhere in the codebase IMO.

With allowedPublicationStatuses I meant 'allowed as set by the dev in the component's prop'. It doesn't need to be changed everywhere. It's just this prop and is separate from the statuses in e.g. useAdminPublications. It's for that reason we're using publicationStatusFilter; because using publicationStatuses everywhere isn't possible due to naming conflicts (+ it's confusing).

I'm all for using different names to clearly indicate where a value is coming from or used. Check e.g. how I used something like 'publicationStatusesForCurrentTab' in the review PR. Such subtle name differences make it much clearer where the value is coming from.

Reply:
> If it's an array of potentially multiple values the name should be plural to indicate this, even if it can be an array of one item. It's how we do it everywhere in the codebase, right?

No, we don't. The variable name should describe what something is. In the `Map` component we have a prop called `centerLatLng`, which is an array of two values. Should we call this `centerLatLngs` too? No, because it is only one pair of lat/long coordinates, not multiple pairs.

In the `Pagination` component we have a variable called `range`, which is an array. Should we call it `ranges`? No, it is only one range, it's just that a range can by definition only be defined as a pair of values.

Conversely, very often we use the variable or attribute name `data` both for objects or arrays. If it's not an array, should we use `datum` instead, since that's singular? In this case I would say no, because `data` is used as an uncountable noun, like 'sand or 'water' or 'stuff'.

>It's for that reason we're using publicationStatusFilter; because using publicationStatuses everywhere isn't possible due to naming conflicts (+ it's confusing).

I meant that there are many more places in the codebase where we use this terminology, most of them related to admin publications as well (see image). If we change it in one place we should change it in all those places. But I do like `publicationStatusesForCurrentTab` instead of `publicationStatuses`, will take those up

![image](https://user-images.githubusercontent.com/13822005/153009151-cb867052-4aee-4a45-9698-f39bc95b8113.png)